src/app/base/components/NetworkActionRow/NetworkActionRow.tsx

```diff
@@ -15,38 +23,64 @@ type Action = {
 };
 
 type Props = {
-  expanded: Expanded | null;
   extraActions?: Action[];
   node: Node;
   rightContent?: ReactNode;
   setExpanded: SetExpanded;
+  selected?: Selected[];
+  setSelected?: SetSelected;
 };
 
 export const NETWORK_DISABLED_MESSAGE =
   "Network can't be modified for this machine.";
 
 const NetworkActionRow = ({
-  expanded,
   extraActions,
   node,
   rightContent,
   setExpanded,
+  selected,
+  setSelected,
 }: Props): JSX.Element | null => {
   const isAllNetworkingDisabled = useIsAllNetworkingDisabled(node);
+  const { setSidePanelContent } = useSidePanel();
+  const { pathname } = useLocation();
+  const isMachinesPage = pathname.startsWith("/machine");
 
   const actions: Action[] = [
     {
-      disabled: [
-        [isAllNetworkingDisabled, NETWORK_DISABLED_MESSAGE],
-        // Disable the button when the form is visible.
-        [expanded?.content === ExpandedState.ADD_PHYSICAL],
-      ],
+      disabled: [[isAllNetworkingDisabled, NETWORK_DISABLED_MESSAGE]],
       label: "Add interface",
       state: ExpandedState.ADD_PHYSICAL,
     },
     ...(extraActions || []),
   ];
 
+  const handleButtonClick = (state: ExpandedState) => {
+    if (state === ExpandedState.ADD_PHYSICAL) {
+      if (isMachinesPage) {
+        setSidePanelContent({
+          view: MachineSidePanelViews.ADD_INTERFACE,
+          extras: { systemId: node.system_id },
+        });
+      } else {
+        setSidePanelContent({ view: DeviceSidePanelViews.ADD_INTERFACE });
+      }
+    } else if (state === ExpandedState.ADD_BOND) {
+      setSidePanelContent({
+        view: MachineSidePanelViews.ADD_BOND,
+        extras: { systemId: node.system_id, selected: selected, setSelected },
+      });
+    } else if (state === ExpandedState.ADD_BRIDGE) {
+      setSidePanelContent({
+        view: MachineSidePanelViews.ADD_BRIDGE,
+        extras: { systemId: node.system_id, selected: selected, setSelected },
+      });
+    } else {
+      setExpanded({ content: state });
+    }
+  };
```

Comment:
This is a good candidate for the lookup table pattern, which I already mentioned some time ago [in a different PR you created](https://github.com/canonical/maas-ui/pull/5189).